### DISCUSSION TITLE
Don't sort directories first

### DISCUSSION THREAD

**Discussion by kirasok** [nvim-tree/nvim-tree.lua#2311] [Q&A] [answered]
Is there way to make nvim-tree *not* to sort directories first?

Like this:
```
a/
a.txt
bcd/
bcd.txt
e.txt
rtx/
```

instead of
```
a/
bcd/
rtx/
a.txt
bcd.txt
e.txt
```

**Comment by chukinas** [comment]
I was just searching for this answer too. I'd love to see exactly what @kirasok is describing.

**Comment by chukinas** [comment]
Love the library though. Use it all the time. Keep up the good work

**Comment by alex-courtis** [comment, marked as answer]
`:h nvim-tree.sort.sort_folders_first` defaults to true

**Reply by chukinas** [reply]
Thanks! But where do I add this? I tried adding it as a sibling to my `sort_by` in `setup`:
```lua
  ...
  sort = {
    sort_folders_first = false
  },
  sort_by = "case_sensitive",
  ...
```

That didn't seem to work.

**Reply by chukinas** [reply]
I think maybe you have a typo in your suggestion - it's `sort.folders_first`, not `sort.sort_folders_first`?

But even removing that errant `sort_` from the above doesn't work. I get a warning about there being no `sort` setup option.

**Reply by alex-courtis** [reply]
Please accept my apologies, the documentation was incorrect. Fixed #2355

**Reply by alex-courtis** [reply]
```lua
  sort = {
    sorter = "case_sensitive",
    folders_first = true,
  },
```

Should function correctly.

`sort_by` was changed to `sort.sorter` however `sort_by` will continue to function.

**Reply by chukinas** [reply]
Yup, that worked! (Plus, I needed to upgrade to a newer version of the library).

Thanks!

**Comment by kirasok** [comment]
Thanks!
